solo/solo.py

```diff
@@ -178,248 +116,132 @@ def main():
+    is_doublet_pred = np.argmin(logit_predictions, axis=1)
+
+    validation_is_doublet_known = is_doublet_known[solo.validation_indices]
+    validation_is_doublet_pred = is_doublet_pred[solo.validation_indices]
+    training_is_doublet_known = is_doublet_known[solo.train_indices]
+    training_is_doublet_pred = is_doublet_pred[solo.train_indices]
+    test_is_doublet_known = is_doublet_known[solo.test_indices]
+    test_is_doublet_pred = is_doublet_pred[solo.test_indices]
+
+    valid_as = accuracy_score(validation_is_doublet_known, validation_is_doublet_pred)
+    valid_roc = roc_auc_score(validation_is_doublet_known, validation_is_doublet_pred)
+    valid_ap = average_precision_score(validation_is_doublet_known, validation_is_doublet_pred)
+
+    train_as = accuracy_score(training_is_doublet_known, training_is_doublet_pred)
+    train_roc = roc_auc_score(training_is_doublet_known, training_is_doublet_pred)
+    train_ap = average_precision_score(training_is_doublet_known, training_is_doublet_pred)
+
+    test_as = accuracy_score(test_is_doublet_known, test_is_doublet_pred)
+    test_roc = roc_auc_score(test_is_doublet_known, test_is_doublet_pred)
+    test_ap = average_precision_score(test_is_doublet_known, test_is_doublet_pred)
+
+    print(f'Training results')
+    print(f'AUROC: {train_roc}, Accuracy: {train_as}, Average precision: {train_ap}')
+
+    print(f'Validation results')
+    print(f'AUROC: {valid_roc}, Accuracy: {valid_as}, Average precision: {valid_ap}')
+
+    print(f'Test results')
+    print(f'AUROC: {test_roc}, Accuracy: {test_as}, Average precision: {test_ap}')
 
     # write predictions
     # softmax predictions
-    order_y, order_score = strainer.compute_predictions(soft=True)
-    _, order_pred = strainer.compute_predictions()
-    doublet_score = order_score[:, 1]
+    softmax_predictions = softmax(logit_predictions, axis=1)
+    doublet_score = softmax_predictions[:, 0]
     np.save(os.path.join(args.out_dir, 'no_updates_softmax_scores.npy'), doublet_score[:num_cells])
     np.savetxt(os.path.join(args.out_dir, 'no_updates_softmax_scores.csv'), doublet_score[:num_cells], delimiter=",")
-
     np.save(os.path.join(args.out_dir, 'no_updates_softmax_scores_sim.npy'), doublet_score[num_cells:])
 
     # logit predictions
-    logit_y, logit_score = logits_strainer.compute_predictions(soft=True)
-    logit_doublet_score = logit_score[:, 1]
+    logit_doublet_score = logit_predictions[:, 0]
     np.save(os.path.join(args.out_dir, 'logit_scores.npy'), logit_doublet_score[:num_cells])
     np.savetxt(os.path.join(args.out_dir, 'logit_scores.csv'), logit_doublet_score[:num_cells], delimiter=",")
-
     np.save(os.path.join(args.out_dir, 'logit_scores_sim.npy'), logit_doublet_score[num_cells:])
 
-
     # update threshold as a function of Solo's estimate of the number of
     # doublets
     # essentially a log odds update
     # TODO put in a function
+    # currently overshrinking softmaxes
```

Comment:
@davek44 this code which modulates the softmax acores currently overshrinks the softmaxes

Reply:
If you point me to an example predictions output, I can help debug

Reply:
an example output: `/home/nicholas/projects/genomics/scrnaseq/doublets/analysis/results_pbmc_1/`

Reply:
@davek44  did you ever take a look at this?

Reply:
Ah no, sorry. I'll queue it